File: Oct1/Lecture11Examples/WeatherViewModel.java
package com.example.example35;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;


public class WeatherViewModel extends AndroidViewModel {
    private MutableLiveData<WeatherData> jsonData;
    private WeatherRepository mWeatherRepository;

    public WeatherViewModel(Application application){
        super(application);
        mWeatherRepository = WeatherRepository.getInstance(application);
        jsonData = mWeatherRepository.getData();
    }
    public void setLocation(String location){

        mWeatherRepository.setLocation(location);
    }

    public LiveData<WeatherData> getData(){
        return jsonData;
    }


}

File: Sep24/Example34/app/src/main/java/com/varunshankar/example34/WeatherViewModel.kt
package com.varunshankar.example34

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.varunshankar.example34.WeatherData
import com.varunshankar.example34.WeatherRepository
import androidx.lifecycle.LiveData

class WeatherViewModel(application: Application) : AndroidViewModel(application) {
    private val jsonData: MutableLiveData<WeatherData>
    private val mWeatherRepository: WeatherRepository
    fun setLocation(location: String?) {
        mWeatherRepository.setLocation(location)
    }

    val data: LiveData<WeatherData>
        get() = jsonData

    init {
        mWeatherRepository = WeatherRepository.getInstance(application)
        jsonData = mWeatherRepository.data
    }
}